rc/extra/just.kak

```diff
@@ -0,0 +1,61 @@
+# Detection
+# ‾‾‾‾‾‾‾‾‾
+
+hook global BufCreate .*/?[jJ]ustfile %{
+    set-option buffer filetype justfile
+}
+
+# Indentation
+# ‾‾‾‾‾‾‾‾‾‾‾
+
+define-command -hidden just-indent-on-new-line %{
+     evaluate-commands -draft -itersel %{
+        # preserve previous line indent
+        try %{ execute-keys -draft \;K<a-&> }
+        # cleanup trailing white spaces on previous line
+        try %{ execute-keys -draft k<a-x> s \h+$ <ret>"_d }
+        # copy '#' comment prefix and following white spaces
+        try %{ execute-keys -draft k <a-x> s ^\h*//\h* <ret> y jgh P }
+    }
+}
+
+# Highlighters
+# ‾‾‾‾‾‾‾‾‾‾‾‾
+
+add-highlighter shared/ regions -default content justfile \
+	comment  '#' '$'  ''                 \
+    double_string '"' (?<!\\)(\\\\)*" '' \
+    single_string "'" (?<!\\)(\\\\)*' '' \
+	inline   '`' '`' ''                  \
+	shell    '^\h+' '^[^\h]' ''
+
+add-highlighter shared/justfile/comment fill comment
+add-highlighter shared/justfile/single_string fill string
+add-highlighter shared/justfile/double_string fill string
+add-highlighter shared/justfile/shell ref sh
+add-highlighter shared/justfile/inline ref sh
+
+#                                                                                    @          recipe     params  :          depends
+add-highlighter shared/justfile/content regex '^(@)?([\w-]+)(?:\s(.+))?\s?(:)(.+)?$' 1:operator 2:function 3:value 4:operator 5:type
+add-highlighter shared/justfile/content regex '([=+])' 1:operator
+add-highlighter shared/justfile/content regex '^([\w-]+)\s=' 1:value
+
+#hook -group justfile-highlight global WinSetOption filetype=justfile %{
+    #add-highlighter window ref justfile
+#}
+
+hook global WinSetOption filetype=justfile %{
+    hook window InsertChar \n -group justfile-indent just-indent-on-new-line
+    add-highlighter window ref justfile
+
+    # hack to add global variable substititution highlighting
+    add-highlighter buffer group justfile
+    add-highlighter buffer/justfile regex '(\{{2})([\w-]+)(\}{2})' 1:operator 2:variable 3:operator
```

Comment:
I think all highlighters should be set on the window, rather than the buffer.

Reply:
This hack allows to highlight `{{VALUE}}` variable substitutions in sh regions. I don't know how to do this better, any ideas?

Reply:
I think `add-highlighter shared/justfile/shell regex […]` should be enough, without need for declaring a hook in the `buffer` scope.

Reply:
Thanks, works like a charm 👍